File: Python2/invert_binary_tree.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution(object):
    def invertTree(self, root):
        self.dfs(root)
        return root

    def dfs(self, root):
        if not root:
            return
        if root.left or root.right:
            temp = root.left
            root.left = root.right
            root.right = temp
        self.dfs(root.left)
        self.dfs(root.right)


if __name__ == '__main__':
    t1 = TreeNode(4)
    t1.left = TreeNode(2)
    t1.right = TreeNode(7)
    t1.left.left = TreeNode(1)
    t1.left.right = TreeNode(3)
    t1.right.left = TreeNode(6)
    t1.right.right = TreeNode(9)

    sol = Solution()
    sol.invertTree(t1)

File: JavaScript/invert_binary_tree.js
// Definition for a binary tree node.
// function TreeNode(val) {
//     this.val = val;
//     this.left = this.right = null;
// }
class TreeNode {
    constructor(val) {
        this.val = val;
        this.left = this.right = null;
    }
}

var invertTree = function(root) {
    dfs(root);
    return root;
}

function dfs(root) {
    if (!root) {
        return;
    }
    if (root.right || root.left) {
        const temp = root.left;
        root.left = root.right;
        root.right = temp;
    }
    dfs(root.right);
    dfs(root.left);
    return;
}

let tree = new TreeNode(4);
tree.left = new TreeNode(2);
tree.left.left = new TreeNode(1);
tree.left.right = new TreeNode(3);
tree.right = new TreeNode(7);
tree.right.left = new TreeNode(6);
tree.right.right = new TreeNode(9);



invertTree(tree);

console.log(tree);